sdk/keyvault/Azure.Security.KeyVault.Keys/src/Key.cs

```diff
@@ -1,40 +1,90 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.Text.Json;
 
 namespace Azure.Security.KeyVault.Keys
 {
     /// <summary>
-    /// Key is the resource consisting of name, <see cref="JsonWebKey"/> properties and its attributes inherited from <see cref="KeyBase"/>.
+    /// <see cref="Key"/> is the resource consisting of a value and its <see cref="Properties"/>.
     /// </summary>
-    public class Key : KeyBase
+    public class Key : IJsonDeserializable, IJsonSerializable
     {
+        private const string KeyPropertyName = "key";
+
+        internal Key()
+        {
+            Properties = new KeyProperties();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Key"/> class.
+        /// </summary>
+        /// <param name="name">The name of the key.</param>
+        public Key(string name)
+        {
+            Properties = new KeyProperties(name);
+        }
+
+        /// <summary>
+        /// Name of the key.
+        /// </summary>
+        public string Name => Properties.Name;
+
+        /// <summary>
+        /// Key identifier.
+        /// </summary>
+        public Uri Id => Properties.Id;
+
+        /// <summary>
+        /// Vault base URL.
+        /// </summary>
+        public Uri VaultUri => Properties.VaultUri;
+
+        /// <summary>
+        /// Version of the key.
+        /// </summary>
+        public string Version => Properties.Version;
+
         /// <summary>
         /// As of http://tools.ietf.org/html/draft-ietf-jose-json-web-key-18
         /// </summary>
         public JsonWebKey KeyMaterial { get; set; }
 
-        internal Key() { }
-
         /// <summary>
-        /// Initializes a new instance of the Key class.
+        /// Gets or sets the attributes of the <see cref="Key"/>.
         /// </summary>
-        /// <param name="name">The name of the key.</param>
-        public Key(string name) : base(name) { }
+        public KeyProperties Properties { get; }
 
-        internal override void ReadProperties(JsonElement json)
+        internal virtual void ReadProperty(JsonProperty prop)
         {
-            KeyMaterial = null;
-
-            if (json.TryGetProperty("key", out JsonElement key))
+            switch (prop.Name)
             {
-                KeyMaterial = new JsonWebKey();
-                KeyMaterial.ReadProperties(key);
-                ParseId(KeyMaterial.KeyId);
+                case KeyPropertyName:
+                    KeyMaterial = new JsonWebKey();
+                    KeyMaterial.ReadProperties(prop.Value);
+                    Properties.ParseId(KeyMaterial.KeyId);
+                    break;
+
+                default:
+                    Properties.ReadProperty(prop);
+                    break;
             }
+        }
+
+        internal virtual void WriteProperties(Utf8JsonWriter json)
```

Comment:
WriteProperties has no implementation? Does the class need to be IJsonSerializable?

Reply:
Only `KeyImportOptions` needed it (didn't call base), so I went ahead and removed this implementation and put it only on `KeyImportOptions`.